### DISCUSSION TITLE
请问CHANGELOG的emoji是怎么生成的？

### DISCUSSION THREAD

**Discussion by hemengke1997** [Tencent/tdesign-react#862] [Q&A]
clone了源码，运行了 `package.json` 中的生成CHANGELOG的命令，但并没有生成emoji，看源码里面没有相关的处理，请问这是怎么实现的啊？ 还有每个changelog item后面 @了开发者，也没看到有代码~
好奇心驱使，期待回答~

**Comment by honkinglin** [comment]
这里代码忘了更新了，早期是本地脚本拉取 git 的 commit 记录然后二次调整的，比较费时费力，现在本地的changelog脚本其实已经废弃了。
目前生成changelog都是流水线自动抓取pr以及结合pr内容来生成的，因为是在github平台所以很容易拿到开发者的信息，具体代码可以参考这里：
https://github.com/Tencent/tdesign-react/blob/develop/.github/workflows/auto-release.yml
